src/test/ui/const-generics/issues/issue-70155.rs

```diff
@@ -0,0 +1,12 @@
+// check-pass
+#![allow(incomplete_features)]
+#![feature(const_generics)]
+
+// Regression test for #70155:
+// `RangeInclusive` should be usable with const generics
+
+struct S<const R: std::ops::RangeInclusive<usize>>;
```

Comment:
Are there tests for the other range types? If so, please place this test with them, if not, add tests for them here, too.

Reply:
`RangeInclusive` is the only range type more complex than just a wrapper around a stable public `start` and/or `end` field, where the equality implementation might reasonably change. 

I thus initially only added this regression test for `RangeInclusive`, but can add them for the other ranges for completeness.